geniza/corpus/models.py

```diff
@@ -533,6 +536,108 @@ def index_data(self):
         # script+language when/if included in index data
     }
 
+    def merge_with(self, merge_docs, rationale, user=None):
+        """Merge the specified documents into this one. Combines all
+        metadata into this document, adds the merged documents into
+        list of old PGP IDs, and creates a log entry documenting
+        the merge, including the rationale."""
+
+        # initialize old pgpid list if previously unset
+        if self.old_pgpids is None:
+            self.old_pgpids = []
+
+        # if user is not specified, log entry will be associated with
+        # script and document will be flagged for review
+        script = False
+        if user is None:
+            user = User.objects.get(username=settings.SCRIPT_USERNAME)
+            script = True
+
+        description_chunks = [self.description]
+        language_notes = [self.language_note] if self.language_note else []
+        notes = [self.notes] if self.notes else []
+        needs_review = [self.needs_review] if self.needs_review else []
+
+        for doc in merge_docs:
+            # add merge id to old pgpid list
+            self.old_pgpids.append(doc.id)
+            # add any tags from merge document tags to primary doc
+            self.tags.add(*doc.tags.names())
+            # add description if set and not duplicated
+            if doc.description and doc.description not in self.description:
+                description_chunks.append(
+                    "Description from %s:\n%s" % (doc.shelfmark, doc.description)
+                )
+            # add any notes
+            if doc.notes:
+                notes.append("Notes from %s:\n%s" % (doc.shelfmark, doc.notes))
+            if doc.needs_review:
+                needs_review.append(doc.needs_review)
+
+            # add languages and probable languages
+            for lang in doc.languages.all():
+                self.languages.add(lang)
+            for lang in doc.probable_languages.all():
+                self.probable_languages.add(lang)
+            if doc.language_note:
+                language_notes.append(doc.language_note)
+
+            # if there are any textblocks with fragments not already
+            # asociated with this document, reassociate
+            # (i.e., for newly discovered joins)
+            # does not deal with discrepancies between text block fields or order
+            for textblock in doc.textblock_set.all():
+                if textblock.fragment not in self.fragments.all():
+                    self.textblock_set.add(textblock)
+
+            # reassociate all footnotes
+            for footnote in doc.footnotes.all():
+                # any good way to check if any are exactly the same?
+                self.footnotes.add(footnote)
```

Comment:
we could implement `__eq__` for `Footnote`. i think the combination of content object, source, relation type, and location would constitute uniqueness

Reply:
Yeah, I thought about that too, but didn't want to take it on right now. I was thinking that at least for now we can ask them to check footnotes when reviewing merged documents. Although implementing it might not be all that hard... (and even if we do they would probably still need to check the footnotes). What do you think?

Reply:
i think we should assume that there may actually be many exact duplicate footnotes, since for a decent number of the merge groups they went to the trouble of ensuring that the description was also duplicated exactly. that being the case, it seems worthwhile to implement `__eq__` so we can do a check here and maybe save some work

Reply:
@thatbudakguy what do you think, should notes and content (i.e. provisional transcription content) be included in footnote equality check?

Reply:
Actually, to use this for the merge check, the simplest thing would be if footnote equality does to _not_ compare content object — but I'm not sure if that's logical for any other use case... What do you think?